connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaStatusBackingStoreFormatTest.java

```diff
@@ -216,7 +219,12 @@ public void putTopicStateRetriableFailure() {
         }).when(kafkaBasedLog).send(eq(key), valueCaptor.capture(), any(Callback.class));
 
         store.put(topicStatus);
-        verify(kafkaBasedLog, times(2)).send(any(), any(), any());
+        try {
+            verify(kafkaBasedLog, times(2)).send(any(), any(), any());
```

Comment:
Did I introduce flakiness with KAFKA-14902?
If yes, I think this verify should be enhanced with a timeout (times(2).timeout(...)) to allow the background thread to trigger the retry, then you don't need to mock the executor

Reply:
@urbandan , that's what I am suspecting. Ok I wanted to verify if this was the case with this hackish fix ( and also was not aware of timeout option). Let me set a timeout of 300ms and see if that helps.

Reply:
@urbandan , thanks that worked!